src/rtpghi~.c

```diff
@@ -131,16 +153,92 @@ void rtpghi_tilde_tol(t_rtpghi_tilde *x, t_floatarg f)
 {
 	if ((f > 0.0) && (f < 1.0)) {
 		double tol = f;
-		phaseret_rtpghi_set_tol_s(x->sta_pd, tol);
+		phaseret_rtpghi_set_tol(x->sta_pd, tol);
 	}
 	else {
 		pd_error(x, "failed to set tol, input float has to be  > 0 && <1, but got: %f", f);
 	}
 }
 
-/*void rtpghi_tilde_res(t_rtpghi_tilde *x)
+void rtpghi_tilde_overlap(t_rtpghi_tilde *x, t_floatarg f)
+{
+	if ((f > 0.0) && (f <= x->blocksize)) {
+		phaseret_rtpghi_done(&(x->sta_pd));
+		post("destroyed state at adrr: %p\n", &(x->sta_pd));
+		x->sta_pd=NULL;
```

Comment:
it's probably better to refactor the (conditional) recreation into a separate function, that checks whether (at least) one of the constituent parameters has changed and (re)creates the `phaseret_rtpghi_state*` object.

something like:
```
void rtpghi_tilde_recreatestate(t_rtpghi_tilde*x, LTFAT_FIRWIN win, int blocksize, int overlap) {
   if(x->sta_pd && (win != x->win || blocksize != x->blocksize || overlap != x->overlap)) {
        phaseret_rtpghi_done(&(x->sta_pd));
        x->sta_pd=NULL;
   }
   x->win = win;
   x->blocksize = blocksize;
   x->overlap = overlap;
   if(!x->sta_pd) {
      if(phaseret_rtpghi_init(x->win, x->overlap, x->blocksize, ...)) {
         pd_error(x, "failed to init state");
          x->sta_pd=NULL;
      }
   }
}